### PR TITLE
Update unsigned suffixes

### PR DESCRIPTION
**Pull request by pfl**
Adding more of the 'U' suffixes to unsigned variable values all over the code base except /ext. Part of #9895.

### CODE REVIEW

**Review by cvinayak — APPROVED** [review]

**Comment by pfl** [comment]
Updated somebody else's code that caused errors.

**Review by mike-scott — APPROVED** [review]
LGTM for LwM2M changes.  @pfl Will these now be caught by PR review?

**Review by tbursztyka — APPROVED** [review]

**Review by aescolar — APPROVED** [review]
+1 for native_posix, nrf52_bsim and their drivers' changes
Have not reviewed the other 550 files ;)

**Review by himanshujha199640 — DISMISSED** [review]
Cc  @JuliaLawall

@pfl 
Could you please check this once:
```
himanshu@himanshu-Vostro-3559:~/zephyr$ spatch -D report scripts/coccinelle/unsigned_suffix.cocci drivers/pwm/pwm_imx.c
init_defs_builtins: /usr/local/bin/../lib/coccinelle/standard.h
HANDLING: drivers/pwm/pwm_imx.c
drivers/pwm/pwm_imx.c:74:7-10: WARNING: Unsigned 'U' suffix missing
himanshu@himanshu-Vostro-3559:~/zephyr$ spatch -D patch scripts/coccinelle/unsigned_suffix.cocci drivers/pwm/pwm_imx.c
init_defs_builtins: /usr/local/bin/../lib/coccinelle/standard.h
HANDLING: drivers/pwm/pwm_imx.c
diff = 
--- drivers/pwm/pwm_imx.c
+++ /tmp/cocci-output-15854-f2f185-pwm_imx.c
@@ -71,7 +71,7 @@ static int imx_pwm_pin_set(struct device
 
 	LOG_DBG("enabled=%d, pulse_cycles=%d, period_cycles=%d,"
 		    " duty_cycle=%d\n", enabled, pulse_cycles, period_cycles,
-		    (100 * pulse_cycles / period_cycles));
+		    (pulse_cycles * 100U / period_cycles));
 
 	/*
 	 * i.MX PWMv2 has a 4-word sample FIFO.
```
The `patch` case didn't workout as multiplication didn't match.
So, I added the following diff on top of your patch to catch it:
```
diff --git a/scripts/coccinelle/unsigned_suffix.cocci b/scripts/coccinelle/unsigned_suffix.cocci
index f63e2883f4..cb48d7b102 100644
--- a/scripts/coccinelle/unsigned_suffix.cocci
+++ b/scripts/coccinelle/unsigned_suffix.cocci
@@ -100,9 +100,8 @@ identifier r_rewrite.z;
 - C
 + z
 |
-  v *
-- C
-+ z
+- v * C
++ v * z
 |
   v /
 - C
```

**Review by ceolin — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]

File: drivers/flash/flash_stm32f0x.c [hunk 1]
```diff
@@ -24,7 +24,7 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 bool flash_stm32_valid_range(struct device *dev, off_t offset, u32_t len,
 			     bool write)
 {
-	return (!write || (offset % 2 == 0 && len % 2 == 0)) &&
+	return (!write || (offset % 2 == 0 && len % 2 == 0U)) &&
```
Comment (erwango) [inline drivers/flash/flash_stm32f0x.c]: Missed a value here ?
```
(offset % 2 == 0U && len % 2 == 0U)
```

**Review by erwango — DISMISSED** [review]
Did some mess this morning, try to set things back.

File: drivers/flash/flash_stm32f0x.c (hunk 1 above)
Comment (jhedberg) [reply drivers/flash/flash_stm32f0x.c]: off_t is signed, AFAIK

Comment (erwango) [reply drivers/flash/flash_stm32f0x.c]: ok, right

**Comment by erwango** [comment]
@kgugala, @MaureenHelm, @rveerama1, @Mani-Sadhasivam, @jfischer-phytec-iot, 
@sjanc, @carlescufi, @galak, @nordic-krch, @anangl, @nvlsianpu, @loicpoulain,
@sathishkuttan, @rgundi , @ramakrishnapallala, @jarz-nordic , @thoh-ot , @pfl,
@dcpleung , @gnuless and @nategraff-sif:
You were initially in review of this PR. I've done some mess that removed you somehow.
Now, I can't add you manually (github does not allow to add more than 15 reviewers manually, it appears). So please consider you are still asked for review.
Sorry for the mess.

**Review by Mani-Sadhasivam — APPROVED** [review]
+1 for LED changes

**Review by vanwinkeljan — APPROVED** [review]
display related changes lgtm

**Review by dcpleung — APPROVED** [review]

**Comment by pfl** [comment]
> So, I added the following diff on top of your patch to catch it:
Added your changes into a new commit and the updated files into another in order to make the additional changes to stand out.

**Pull request merged**